src/color.rs

```diff
@@ -430,6 +431,112 @@ fn color_f32_to_u8(
     ]
 }
 
+/// The colorspace used to interpret pixel values.
+///
+/// This is a very limited subset of SkColorSpace.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Debug)]
+pub enum ColorSpace {
+    /// Linear RGB, the default.  Assumes #7f7f7f is half as bright as #ffffff.
+    Linear,
+
+    /// Apply a gamma factor of 2.
+    ///
+    /// This is the fastest gamma correction to apply, and produces reasonable
+    /// quality blending.
+    Gamma2,
+
+    /// Apply a gamma factor of 2.2.
+    ///
+    /// This is also known as "simple sRGB"; it's very close to the actual gamma
+    /// function specified by the sRGB color space, but much faster to compute.
+    SimpleSRGB,
+
+    /// Apply the full sRGB gamma function.
+    ///
+    /// This does not convert the RGB colors to CIE XYZ for blending; it only
+    /// applies the (full) gamma function.
+    FullSRGBGamma,
+}
+
+impl Default for ColorSpace {
+    fn default() -> Self {
+        ColorSpace::Linear
+    }
+}
+
+impl ColorSpace {
+    pub(crate) fn expand_channel(self, x: NormalizedF32) -> NormalizedF32 {
+        match self {
+            ColorSpace::Linear => x,
+            ColorSpace::Gamma2 => x * x,
+            ColorSpace::SimpleSRGB => {
+                NormalizedF32::new_clamped(approx_powf(x.get(), 2.2))
+            }
+            ColorSpace::FullSRGBGamma => {
+                let x = x.get();
+                let x = if x <= 0.04045 {
+                    x / 12.92
+                } else {
+                    approx_powf((x + 0.055) / 1.055, 2.4)
+                };
+                NormalizedF32::new_clamped(x)
+            }
+        }
+    }
+
+    pub(crate) fn expand_color(self, mut color: Color) -> Color {
+        color.r = self.expand_channel(color.r);
```

Comment:
We could use SIMD `approx_powf` here. Mainly to prevent code duplication. 

Or we could simply use the default/std `powf`, since this is not a performance sensitive part of code.

Reply:
The CI test run implied that the standard `powf` is not available in no-std environments.  In addition, since we only have three values to scale here, using the 8-wide SIMD implementation is likely to decrease performance.